text/go/0001-remove-dep.md

```diff
@@ -0,0 +1,26 @@
+# Remove the `dep` and `dep-ensure` Buildpacks
+
+## Proposal
+
+The `dep` and `dep-ensure` Buildpacks should be removed.
+
+## Motivation
+
+`dep` was a package management tool created for Go before the introduction of
+`go mod`. Since the introduction of `go mod` the `dep` tool has since been
+deprecated and archived by the team working on it (See the [Github
+Repository](https://github.com/golang/dep) and the [`dep`
+documentation](https://golang.github.io/dep/docs/introduction.html)). Because
+this tool has been deprecated and `go mod` has been widely adopted by the Go
+community, we should remove our buildpacks.
+
+## Implementation
```

Comment:
I would like to see the buildpack help users migrate. For example, what if we shipped a version of the buildpack that still worked but printed loud deprecation warnings and a link to migration steps if it detected/ran? We could include that version in the language family buildpack for a period of time before removing it.

If we just remove it without warning that it feels a bit like we have failed to provide a deprecation window, which isn't the best experience for users, even if the tool itself is deprecated.

Reply:
I think a deprecation period a good idea. A month or so seems reasonable, given the underlying tech is already out of support.

Reply:
Ok I have specified that a 30 day deprecation window should be implemented.